src/model.js

```diff
@@ -310,6 +310,14 @@ export default class MediaPlayerObject {
         return this.callService(e, `${platform}_UNJOIN`, options);
       }
       if (platform === 'soundtouch') {
+        // if no entity is specified, remove self (leave group)
+        if (!entity || !entity.length) {
+          entity = this.id;
+        }
+        // don't send master entity as slave to remove
+        if (typeof entity == "object" && entity.indexOf(this.master) > -1) {
+          entity.splice(entity.indexOf(this.master), 1);
+        }
```

Comment:
Do we need to do this?

Reply:
still debugging this. For some reason it doesn't seem to work when I send multiple slaves to the `remove_zone_slave` service/method, even though the HA component as well as libsoundtouch seem to support this. When I click the button in the player, nothing happens. When trigger the service manually with the according payload (containing all slaves) just one is actually removed. The other one won't ever get removed no matter how often I click or regardless which order they have in the payload. Something really odd seems to be going on here.
I'll try to send the HTTP POST request to the master manually in the next days to see if the issue is in the API of BOSE or somewhere in our call chain

Reply:
I think this works for me. I'll have to double-check tonight when I am home.
But I am using libsoundtouch 0.8.0 base rather than what you have, and I do think that the 0.7.2 has some issues like this. This might be a known issue to fix, or you might want to call the libsoundtouch routine repeatedly for each slave to work around it.

Reply:
Just cheked.
Group All does work.
Ungroup does not work.

Reply:
This is the debugging in my logs, which I would need to compare to the BOSE API
```
2019-10-30 20:28:06 INFO (SyncWorker_12) [custom_components.soundtouch.media_player] Removing slaves media_player.bose_stairs, media_player.bose_lounge, media_player.bose_kitchen from zone with master media_player.bose_kitchen
2019-10-30 20:28:06 DEBUG (SyncWorker_12) [libsoundtouch.device] request.post url=http://192.168.2.180:8090/removeZoneSlave, payload=<?xml version="1.0" encoding="UTF-8" ?><zone master="000C8AC06D11"><member ipaddress="192.168.2.182">A81B6A53F41B</member><member ipaddress="192.168.2.181">08DF1F0DC6E0</member><member ipaddress="192.168.2.180">000C8AC06D11</member></zone>, response=<?xml version="1.0" encoding="UTF-8" ?><status>/removeZoneSlave</status>

```

Reply:
yea, it's really odd why it doesn't work. As far as I could trace it back with libsoundtouch 0.7.x, on our end all the info is sent to the API but it doesn't work.
If I find the time I will just fire up wireshark and do some network sniffing to find out what the Soundtouch Android app is sending when destroying a zone.

Reply:
That log I posted above comes from my enhanced libsoundtouch code where am trying to debug the BOSE API behaviour. I keep the response of every API command, so you don't need wireshark or anything.

`response=<?xml version="1.0" encoding="UTF-8" ?><status>/removeZoneSlave</status>
`

=> As I read the API documents, Bose thinks the command was just fine. It didnt do it though!

Reply:
OK, I got the UNGROUP button working.
Here is my code for soundtouch remove_zone_slave.
Note that the commented code seems correct to me, BOSE API acknowledges it but does not execute it.
It was important not to call remove slave on the master, and mini-card does return the master in the list of things to ungroup. So I filter that out. 

It tests well for me.

```
 def remove_zone_slave(self, slaves):
        """
        Remove slave(s) from and existing zone (multi-room).

        Zone must already exist and slaves array can not be empty.
        Note: If removing last slave, the zone will be deleted and you'll have
        to create a new one. You will not be able to add a new slave anymore

        :param slaves: slaves to remove from the zone. Slaves here are of class SoundTouchDevice(MediaPlayerDevice)

        """
        if not slaves:
            _LOGGER.warning("Unable to find slaves to remove")
        else:
            # Underlying BOSE API does not seem to work corrently when removeslave command is sent
            # with multiple slaves. API responds with an OK, it just doesn't do it.
            # So unroll the command into multiple API calls, one for each slave
            #slave_names = [s.entity_id for s in slaves if (slave.entity_id != self.entity_id) ]
            #_LOGGER.info("Removing slaves %s from zone with master %s",
            #             ", ".join(slave_names), self.entity_id)
            #self._device.remove_zone_slave([slave.device for slave in slaves if (slave.entity_id != self.entity_id)])
            for s in slaves:
                if (s.entity_id != self.entity_id):
                    # Only remove slaves, do not call remove removeSlave on itself
                    _LOGGER.info("Removing slave %s from zone with master %s",s.entity_id, self.entity_id)
                    self._device.remove_zone_slave([s.device])

```

Reply:
thanks. IIRC I had already tested removing the slaves one by one, but also didn't work. Will give it another try then

Reply:
nope, not working here. I see in the log that it tries to remove one out of two radios, but nothing actually happens, which is super weird since the "LEAVE" button is working just fine.

Reply:
Hmm, its definitely working for me with the code above.
Here are my detailed debug logs
You are filtering out the master? Calling remove slave with the master as the <member> data caused Bose to do some weird things.

```
2019-11-03 09:52:12 DEBUG (SyncWorker_10) [custom_components.soundtouch.media_player] service_handle <ServiceCall media_player.soundtouch_remove_zone_slave (c:e4cf862899a74c1aad3a253c188245e1): master=media_player.bose_kitchen, slaves=['media_player.bose_kitchen', 'media_player.bose_stairs', 'media_player.bose_lounge', 'media_player.bose_family_room']> with master=media_player.bose_kitchen, slave_ids=media_player.bose_kitchen ,media_player.bose_stairs ,media_player.bose_lounge ,media_player.bose_family_room
2019-11-03 09:52:12 INFO (SyncWorker_10) [custom_components.soundtouch.media_player] Removing slave media_player.bose_stairs from zone with master media_player.bose_kitchen
2019-11-03 09:52:12 DEBUG (SyncWorker_10) [libsoundtouch.device] request.post url=http://192.168.2.180:8090/removeZoneSlave, payload=<?xml version="1.0" encoding="UTF-8" ?><zone master="000C8AC06D11"><member ipaddress="192.168.2.182">A81B6A53F41B</member></zone>, response=<?xml version="1.0" encoding="UTF-8" ?><status>/removeZoneSlave</status>
2019-11-03 09:52:12 INFO (SyncWorker_10) [custom_components.soundtouch.media_player] Removing slave media_player.bose_lounge from zone with master media_player.bose_kitchen
2019-11-03 09:52:13 DEBUG (SyncWorker_10) [libsoundtouch.device] request.post url=http://192.168.2.180:8090/removeZoneSlave, payload=<?xml version="1.0" encoding="UTF-8" ?><zone master="000C8AC06D11"><member ipaddress="192.168.2.181">08DF1F0DC6E0</member></zone>, response=<?xml version="1.0" encoding="UTF-8" ?><status>/removeZoneSlave</status>
2019-11-03 09:52:13 INFO (SyncWorker_10) [custom_components.soundtouch.media_player] Removing slave media_player.bose_family_room from zone with master media_player.bose_kitchen
2019-11-03 09:52:13 DEBUG (SyncWorker_10) [libsoundtouch.device] request.post url=http://192.168.2.180:8090/removeZoneSlave, payload=<?xml version="1.0" encoding="UTF-8" ?><zone master="000C8AC06D11"><member ipaddress="192.168.2.183">B0D5CCD8F1C5</member></zone>, response=<?xml version="1.0" encoding="UTF-8" ?><status>/removeZoneSlave</status>

```

Reply:
looks like I had a bug in my player JS code. Cleaned everything up and it's working now.